podcasts/Chapters.swift

```diff
@@ -1,60 +1,83 @@
-//
-//  Chapters.swift
-//  podcasts
-//
-//  Created by Maarut Chandegra on 11/11/2022.
-//  Copyright © 2022 Shifty Jelly. All rights reserved.
-//
-
-import Foundation
 import UIKit
 
 class Chapters: Equatable {
     private var chapters = [ChapterInfo]()
 
-    init(chapters: [ChapterInfo]) {
-        self.chapters = chapters
-    }
-
-    func visibleChapter() -> ChapterInfo? {
+    var visibleChapter: ChapterInfo? {
         chapters.last(where: { !$0.isHidden })
     }
 
-    func title() -> String {
-        chapters.last?.title ?? ""
-    }
-
-#if !os(watchOS)
-    func artwork() -> UIImage? {
-        chapters.last(where: { $0.image != nil })?.image
+    var title: String {
+        chapters.last(where: { !$0.title.isEmpty })?.title ?? ""
     }
-#endif
 
-    func count() -> Int {
+    var count: Int {
         chapters.count
     }
 
-    func index() -> Int {
-        visibleChapter()?.index ?? -1
+    var index: Int {
+        visibleChapter?.index ?? -1
     }
 
-    func url() -> String? {
+    var url: String? {
         chapters.last(where: { $0.url != nil })?.url
     }
 
-    func startTime() -> CMTime {
-        visibleChapter()?.startTime ??
+    var startTime: CMTime {
+        visibleChapter?.startTime ??
         chapters.min(by: { $0.startTime < $1.startTime} )?.startTime ??
         CMTime()
     }
 
-    func duration() -> TimeInterval {
-        visibleChapter()?.duration ??
+    var duration: TimeInterval {
+        visibleChapter?.duration ??
         chapters.max(by: { $0.duration < $1.duration })?.duration ??
         1
     }
 
+#if !os(watchOS)
+    var artwork: UIImage? {
+        chapters.last(where: { $0.image != nil })?.image
+    }
+#endif
+
+    init() {
+        self.chapters = []
+    }
+
+    init?(chapters: [ChapterInfo]) {
+        if !chaptersOverlap(chapters) {
+            return nil
+        }
```

Comment:
Since the `chapters` property already sets itself to an empty array we can remove the need for an optional and just only set the property if the overlap check succeeds like so:

```suggestion
    init(chapters: [ChapterInfo] = []) {
        if chaptersOverlap(chapters) {
            self.chapters = chapters
        }
    }
```

Reply:
Adopted suggested code.